pubnub-gson/pubnub-gson-impl/src/main/java/com/pubnub/internal/endpoints/presence/SetStateImpl.java

```diff
@@ -86,7 +88,19 @@ protected EndpointInterface<Object> createAction() {
         } else {
             // Some clients require alternative way of setting it through Heartbeat
             // Which is a feature brought over from the legacy Java SDK, and we need to be compatible:
-            return (EndpointInterface<Object>) (Object) new HeartbeatEndpoint(pubnub, channels, channelGroups, state);
+            return (EndpointInterface<Object>) (Object) new HeartbeatEndpoint(pubnub, channels, channelGroups, composeStateParamValue());
         }
     }
+
+    private Object composeStateParamValue() {
+        Map<String, Object> stateParamValue = new HashMap<>();
+        for (String channel : channels) {
+            stateParamValue.put(channel, state);
+        }
+        for (String channelGroup : channelGroups) {
```

Comment:
only channels are allowed in the map, 

maybe we should even check it in `validateParams` that no channel groups where passed if `withHeartbeat` is `true`